pkg/mimirpb/compat.go

```diff
@@ -195,6 +238,106 @@ func fromSpansProtoToSpans(s []BucketSpan) []histogram.Span {
 	return spans
 }
 
+func FromHistogramToHistogramProto(timestamp int64, h *histogram.Histogram) Histogram {
```

Comment:
We should check (and panic or return error) if `h` is ingester histogram here.

Same in `FromFloatHistogramToHistogramProto`.

Reply:
The tsdb internal type "histogram.Histogram" cannot be a float histogram, it is the type in the return value that can be either (the type used for remote write). So no reason to add the check.

Reply:
oh, but it does make sense the other way around in FromHistogramProtoToHistogram, fixing with panic